doc/design/embargo-mvp.md

```diff
@@ -0,0 +1,187 @@
+If so, the public blob will be used instead of uploading the data again to the embargo bucket.
+
+# Data download
+Downloading embargoed assets is the same as downloading normal assets.
+The `/assets/{asset_id}/download` will verify that the user has permission on the embargoed dandiset if the asset is embargoed, and return a presigned URL if so.
+
+Downloading embargoed zarr archives will be done through a special API surface.
+Clients will use zarr's builtin [fsspec](https://zarr.readthedocs.io/en/stable/tutorial.html#io-with-fsspec) store [HTTPFileSystem](https://filesystem-spec.readthedocs.io/en/latest/api.html#fsspec.implementations.http.HTTPFileSystem) like this:
+```
+zarr_archive = zarr.open('https://api.dandiarchive.org/api/zarr/{zarr_id}/files/')
+```
+`https://api.dandiarchive.org/api/zarr/{zarr_id}/files/{path_to_file_or_directory}` will return either a directory listing parsable by HTTPFileSystem, or a redirect to a presigned S3 URL that can be used to download the file.
+Permissions will also be checked just as when doing an asset download.
+
+A test implementation can be found [here](https://github.com/dandi/dandi-api/commit/46f637c4d29d69d34a1a78d820193402d3fc3778).
+
+
+# Django changes
+
+## Models
+The `Dandiset` model will have an `embargo_status` field that is one of `EMBARGOED`, `RELEASING`, or `OPEN`.
+* `OPEN` means that the Dandiset is publically accessible and publishable.
+  This is the state all Dandisets currently have.
+* `EMBARGOED` means that the Dandiset is embargoed.
+  It is not searchable or viewable to non-owners.
+  It is not publishable.
+  Manifest YAML/JSON files will be written to the embargo bucket rather than the public bucket.
+* `RELEASING` means that the Dandiset is currently transitioning from embargoed to public.
+  All modification operations will return a 400 error while the release is in progress.
+  This includes metadata changes and uploads.
+
+A new `EmbargoedAssetBlob` model will be added.
+This behaves the same as a normal `AssetBlob`, but points to the embargo bucket rather than the public bucket.
+It also contains a reference to the `Dandiset` it belongs to.
+
+`Asset`s will point to exactly one `AssetBlob` or `EmbargoedAssetBlob`.
+An `Asset` will be considered "embargoed" if it has an `EmbargoedAssetBlob`.
+An embargoed `Asset` can only belong to the dandiset named in the `EmbargoedAssetBlob`.
+The automatically populated `Asset` metadata will also set the `access` field to the appropriate value depending on whether or not the asset is embargoed.
+
+A new `EmbargoedUpload` model will be added.
+This is just like the normal `Upload` model, but the uploaded data will be sent to the embargoed bucket instead of the public bucket, and the `Upload` will create an `EmbargoedAssetBlob` when finalized.
+It will also contain a reference to the `Dandiset` it belongs.
+
+A new `EmbargoedZarrArchive` model will be added in the same vein as `EmbargoedAssetBlob`.
```

Comment:
should we use `*ZarrFile` terms instead of `*ZarrArchive` to stay inline with below `EmbargoedZarrUploadFile` etc?

Reply:
I think there's a meaningful distinction: the "archive" is the full asset, while the "file" refers to a single file of the zarr (as it is uploaded). @dchiquito?

Reply:
eh... I always disliked referring to `.zarr/` directory as "zarr file" but it grew on me I guess.
Also as we refer to "files" as "blobs", may be then it still could be `ZarrFile` (the "archive") and `ZarrBlob` (an individual file within Zarr)?  or should we may be check how zarr calls itself and individual files? ;)

Reply:
>  referring to .zarr/ directory as "zarr file"
That's the opposite of what we're doing, the entire zarr directory is the "zarr archive", the files it contains are "zarr files".

Looks like zarr might refer to the component files as "chunks", but I'm not sure if that might also refer to subdirectories.